CMakeLists.txt

```diff
@@ -660,12 +658,6 @@ if (BUILD_TESTING)
                 -fsanitize=fuzzer -lstdc++
             )
 
-            # Set the output directory for the fuzzing binaries
-            set(FUZZ_BIN_DIR "${CMAKE_CURRENT_SOURCE_DIR}/tests/fuzz")
-            set_target_properties(${TEST_NAME} PROPERTIES
-                RUNTIME_OUTPUT_DIRECTORY ${FUZZ_BIN_DIR}
-            )
-
             add_test(NAME ${TEST_NAME}
                     COMMAND ${CMAKE_COMMAND} -E env
                         DYLD_LIBRARY_PATH=${CMAKE_CURRENT_BINARY_DIR}/lib:${CMAKE_CURRENT_BINARY_DIR}/tests/testlib:${CMAKE_CURRENT_SOURCE_DIR}/libcrypto-root/lib:$ENV{DYLD_LIBRARY_PATH}
```

Comment:
Making all of the cmake stuff a little more idomatic is great!

While we're at it, I suspect that we can also remove the `DYLD_LIBRARY_PATH` and the `LD_LIBRARY_PATH`? Or at least let's document why we need them.